kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/EncryptionDekCache.java

```diff
@@ -99,11 +100,14 @@ private void afterCacheEviction(@Nullable CacheKey<K> cacheKey,
     /**
      * Obtain a Dek for the KEK in the given {@code encryptionScheme},
      * generating a new one if necessary.
+     *
      * @param encryptionScheme The KEK to get a DEK for.
-     * @return A stage that completes with the DEK.
+     * @param filterThreadExecutor The filter thread executor.
+     * @return A stage that completes on the filter thread with the DEK.
      */
-    public @NonNull CompletionStage<Dek<E>> get(@NonNull EncryptionScheme<K> encryptionScheme) {
-        return dekCache.get(cacheKey(encryptionScheme));
+    public @NonNull CompletionStage<Dek<E>> get(@NonNull EncryptionScheme<K> encryptionScheme,
+                                                @NonNull FilterThreadExecutor filterThreadExecutor) {
```

Comment:
Another change I made in this PR is to pass the `FilterThreadExecutor` into the caches so they can complete the future on a well-defined thread, rather than exposing callers to whatever threads Caffeine is completing things on. This felt to me like a stronger contract. I can't pass the `FilterThreadExecutor` to the cache constructor because we don't have any filter threads at that point.

Reply:
> Another change I made in this PR is to pass the FilterThreadExecutor into the caches so they can complete the future on a well-defined thread, rather than exposing callers to whatever threads Caffeine is completing things on. This felt to me like a stronger contract.

Yeah I like that property its worth doing.

>  I can't pass the FilterThreadExecutor to the cache constructor because we don't have any filter threads at that point.

Is that pointing a general weakness in our object lifecycle? Should we be offering something like an initialisation phase? (Thoughts for the future not this PR)

Reply:
Ohh right, I didn't initially twig that this was now being passed on the get call. I can't think of an alternative solution as each filter is potentially going to be bound to a different executor so I think it has to manifest on the get call. 

One only alliterative is to give each filter a dedicated instance of a wrapper around the cache which supplies the executor when  accessing the shared cache.